pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -87,20 +87,55 @@ type MutableCluster struct {
 	httpProtocolOptions *http.HttpProtocolOptions
 }
 
+// MetadataCerts hosts certificate related metadata specified in proxy metadata.
+type MetadataCerts struct {
+	// tlsServerCertChain is the absolute path to server cert-chain file
+	tlsServerCertChain string
+	// tlsServerKey is the absolute path to server private key file
+	tlsServerKey string
+	// tlsServerRootCert is the absolute path to server root cert file
+	tlsServerRootCert string
+	// tlsClientCertChain is the absolute path to client cert-chain file
+	tlsClientCertChain string
+	// tlsClientKey is the absolute path to client private key file
+	tlsClientKey string
+	// tlsClientRootCert is the absolute path to client root cert file
+	tlsClientRootCert string
+}
+
 // ClusterBuilder interface provides an abstraction for building Envoy Clusters.
 type ClusterBuilder struct {
-	proxy *model.Proxy
-	req   *model.PushRequest
-	cache model.XdsCache
+	proxy         *model.Proxy
+	req           *model.PushRequest
+	cache         model.XdsCache
+	metadataCerts MetadataCerts
+	clusterID     string
 }
 
 // NewClusterBuilder builds an instance of ClusterBuilder.
 func NewClusterBuilder(proxy *model.Proxy, req *model.PushRequest, cache model.XdsCache) *ClusterBuilder {
```

Comment:
in EDS "EndpointBuilder" (equivilent here) *is* the cache key. This makes it ~impossible to misuse. We don't have to do it in this PR but I think thats the best long term direction

Reply:
Yes. I agree. We will have to move to that model and remove proxy altogether.